### PR TITLE
VAGOV-TEAM-90576: Render Employment History list & loop pattern

### LINKED ISSUE #90576: Render Employment History list & loop pattern
- Blocked by #96075.

Render the "List & Loop" pattern based off of the normalized chapter created in #93614. The List & Loop will contain fields specific to the Employment History section of VA Form 21-4140. Also update the "VA Form 21-4140" mock data in `vets-website` to contain a chapter/step for "Employment History" based off of this pattern.

## Acceptance Criteria

 - [ ] **Scenario: Employment History list & loop pattern rendering**
  - Given the List & Loop pattern is implemented
  - When the pattern is used in the Employment History section of VA Form 21-4140
  - Then the pattern should display fields specific to the Employment History section, and the data should be formatted correctly

### PR DESCRIPTION
**Pull request by derekhouck**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Renders the ["List & Loop" pattern](https://design.va.gov/patterns/ask-users-for/multiple-responses) from normalized data created by department-of-veterans-affairs/content-build#2350.
- Also refactors `formatChapter` to standardize behavior among Digital Form patterns. They now all return `FormConfigPages`.

## Related issue(s)

- Closes department-of-veterans-affairs/va.gov-team#90576

## Testing done

- New unit tests for `formatPages` and `listLoopPages`
- Local testing using mock data

## Screenshots

The summary page rendered on form 21-4140:
Screenshot 2024-11-26 at 11 10 26 AM

Name and address page rendered on form 21-4140:
Screenshot 2024-11-26 at 11 12 16 AM

Dates employed page rendered on form 21-4140:
Screenshot 2024-11-26 at 11 13 41 AM

Employment detail page rendered on form 21-4140:
Screenshot 2024-11-26 at 11 14 45 AM

Employment History rendered on the review page for form 21-4140:
Screenshot 2024-11-26 at 11 16 54 AM

## What areas of the site does it impact?

All forms created using the `simple-forms-form-engine` shared code.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

I added a `FormConfigPages` type to make type-hinting easier on the digital form patterns. If there's some official process for adding a new type that I skipped over, let me know.

### CODE REVIEW

**Review by rmessina1010 — DISMISSED** [review]
LGTM

**Review by ryguyk — DISMISSED** [review]
This looks great. Quite impressive, and quite exciting that we're rendering almost the entire form now!

Very nicely done.

File: src/applications/simple-forms-form-engine/shared/tests/unit/utils/digitalFormPatterns.unit.spec.js [hunk 1]
```diff
@@ -2,191 +2,361 @@ import { expect } from 'chai';
...
+      const schemas = addressPages(includeMilitary);
 
       expect(addressSpy.calledOnce).to.eq(true);
-      expect(schemas.uiSchema.address).to.not.eq(undefined);
+      expect(schemas[includeMilitary.id].uiSchema.address).to.not.eq(undefined);
     });
   });
 
   context('when militaryAddressCheckbox is false', () => {
+    let noMiliarySpy;
```
[the remark this answers is not in the document]
Comment (derekhouck) [reply src/applications/simple-forms-form-engine/shared/tests/unit/utils/digitalFormPatterns.unit.spec.js]: I care. Fixed.

**Review by ryguyk — APPROVED** [review]
Thanks for the fix.

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**